### PR TITLE
PSD-1044 repeat step error

### PR DESCRIPTION
**Pull request by MatthewBurstein**
## Checklist:

- [X] Automated checks are passing locally.

### CODE REVIEW

**Comment by jasiekmiko** [comment]
I'll take this one over :)

File: psd-web/app/views/investigations/ts_investigations/_repeatable_info.html.slim [hunk 1]
```diff
@@ -14,14 +14,11 @@
...
-      = render "components/govuk_radios",
-              name: further_key,
-              errorMessage: error_text && { text: error_text },
+      = render "form_components/govuk_radios",
+              form: form, key: further_key,
               fieldset: { legend: { text: other_text, classes: "govuk-fieldset__legend--m" } },
-              items: [{ text: "Yes", value: "Yes", checked: @repeat_step == "Yes",
-                      id: "#{further_key}_yes" },
-                      { text: "No", value: "No", checked: @repeat_step == "No", id: "#{further_key}_no" }]
+              items: [{ text: "Yes", value: "Yes", checked: @repeat_step == "Yes" },
```
Comment (broder) [inline psd-web/app/views/investigations/ts_investigations/_repeatable_info.html.slim]: Are we now setting `@repeat_step` to be `true` / `false` rather that `"Yes"` / `"No"`? https://github.com/UKGovernmentBEIS/beis-mspsds/pull/836/files#diff-46445ecbbea8ff13f67c7ac5d671a650R147

File: psd-web/app/controllers/investigations/ts_investigations_controller.rb [hunk 2]
```diff
@@ -331,36 +338,58 @@ def store_repeat_step
...
-    else
+    end
+  end
+
+  def repeat_step_valid?(model)
+    if @repeat_step.nil?
       further_page_type = to_item_text(step)
-      @investigation.errors.add(further_key(step), "Select whether or not you have #{further_page_type} to record")
+      model.errors.add(further_key(step), "Select whether or not you have #{further_page_type} to record")
+      false
```
Comment (broder) [inline psd-web/app/controllers/investigations/ts_investigations_controller.rb]: Do we need to do `return false` here?

Comment (jasiekmiko) [reply psd-web/app/controllers/investigations/ts_investigations_controller.rb]: Ahh, thanks! I've first put a `return` on both `false` and `true`, then figured I don't need it on `true and didn't think any better after that so removed from both 🤣

File: psd-web/app/views/investigations/ts_investigations/_repeatable_info.html.slim (hunk 1 above)
Comment (jasiekmiko) [reply psd-web/app/views/investigations/ts_investigations/_repeatable_info.html.slim]: Good spot! Fix proposed in https://github.com/UKGovernmentBEIS/beis-mspsds/pull/895

**Pull request merged**